src/Parse/ParseUser.php

```diff
@@ -164,19 +164,13 @@ public static function logInWithFacebook($id, $access_token, $expiration_date =
             $expiration_date = new \DateTime();
             $expiration_date->setTimestamp(time() + 86400 * 60);
         }
-        $data = ['authData' => [
-            'facebook' => [
-                'id'              => $id, 'access_token' => $access_token,
-                'expiration_date' => ParseClient::getProperDateFormat($expiration_date),
-            ],
-        ]];
-        $result = ParseClient::_request('POST', 'users', '', json_encode($data));
-        $user = ParseObject::create('_User');
-        $user->_mergeAfterFetch($result);
-        $user->handleSaveResult(true);
-        ParseClient::getStorage()->set('user', $user);
+        $authData = [
+            'id'              => $id,
+            'access_token'    => $access_token,
+            'expiration_date' => ParseClient::getProperDateFormat($expiration_date),
+        ];
```

Comment:
This looks good. Looking at a coverage report you're only accounting for whether the id or access token is missing, can you beef up the test coverage by running it with some invalid credentials? Ideally you want it to go all the way through one run in at least one your tests.

Running all the way through with bad credentials should throw a `ParseException` with a message of `'Facebook auth is invalid for this user.'`, just so you know what you're expecting there.